python/shotgun_fields/shotgun_field_manager.py

```
@@ -0,0 +1,243 @@
+
+        :returns: The subset of the field_names array that has associated widget classes.
+        """
+        supported_fields = []
+
+        # build a list of all the fields whose type is in the field map
+        for field_name in field_names:
+            data_type = shotgun_globals.get_data_type(sg_entity_type, field_name)
+            if data_type in self.__FIELD_TYPE_MAP:
+                supported_fields.append(field_name)
+
+        return supported_fields
+
+    def create_widget(self, sg_entity_type, field_name, entity=None, parent=None, **kwargs):
+        """
+        Returns the widget class associated with the field type if it has been registered.
+
+        If the entity is passed in and has the value for the requested field in it then the
+        initial contents of the widget will display that value.
+
+        Any keyword args other than those below will be passed to the constructor of whatever
+        QWidget the field widget wraps.
+
+        :param sg_entity_type: Shotgun entity type
+        :type sg_entity_type: String
+
+        :param field_name: Shotgun field name
+        :type field_name: String
+
+        :param entity: The Shotgun entity dictionary to pull the field value from.
+        :type entity: Whatever is returned by the Shotgun API for this field
+
+        :param parent: Parent widget
+        :type parent: :class:`PySide.QtGui.QWidget`
+
+        :returns: :class:`~PySide.QtGui.QWidget` or None if the field type has no display widget
+        """
+        data_type = shotgun_globals.get_data_type(sg_entity_type, field_name)
+
+        if data_type in self.__FIELD_TYPE_MAP:
+            cls = self.__FIELD_TYPE_MAP[data_type]
+
+            # instantiate the widget
+            return cls(
+                parent=parent,
+                entity=entity,
+                field_name=field_name,
+                bg_task_manager=self._task_manager,
+                **kwargs
+            )
+        return None
```

Personal preference, but I would probably raise instead of returning None.